app/src/main/java/net/bible/service/db/readingplan/ReadingPlanDatabaseDefinition.kt

```diff
@@ -0,0 +1,499 @@
+                ${ReadingPlanMeta.COLUMN_DESCRIPTION} TEXT NOT NULL,
+                ${ReadingPlanMeta.COLUMN_DATE_START} INTEGER,
+                ${ReadingPlanMeta.COLUMN_DAYS_IN_PLAN} INTEGER,
+                ${ReadingPlanMeta.COLUMN_CURRENT_DAY} INTEGER NOT NULL DEFAULT 0,
+                ${ReadingPlanMeta.COLUMN_VERSIFICATION_NAME} TEXT NOT NULL
+                ); """
+
+            private const val SQL_CREATE_ENTRIES_DAYS =
+                """CREATE TABLE ${ReadingPlanDays.TABLE_NAME} (
+                ${ReadingPlanDays.COLUMN_ID} INTEGER PRIMARY KEY AUTOINCREMENT,
+                ${ReadingPlanDays.COLUMN_READING_PLAN_META_ID} INTEGER NOT NULL,
+                ${ReadingPlanDays.COLUMN_DAY_NUMBER} INTEGER NOT NULL,
+                ${ReadingPlanDays.COLUMN_READING_DATE} TEXT,
+                ${ReadingPlanDays.COLUMN_DAY_CHAPTERS} TEXT NOT NULL,
+                ${ReadingPlanDays.COLUMN_READ_STATUS} TEXT
+                );"""
+
+            private const val SQL_CREATE_DAYS_INDEX =
+                "CREATE INDEX days_date_name ON ${ReadingPlanDays.TABLE_NAME}(${ReadingPlanDays.COLUMN_READING_DATE});"
+
+            private const val VERSIFICATION = "Versification"
+            private const val DOT_PROPERTIES = ".properties"
+            private const val READING_PLAN_FOLDER = SharedConstants.READINGPLAN_DIR_NAME
+            private val USER_READING_PLAN_FOLDER = SharedConstants.MANUAL_READINGPLAN_DIR
+
+
+            /** Called from CommonDatabaseHelper when ReadingPlan tables
+             * don't exists and the helper class needs to create them.
+             */
+            @JvmStatic
+            fun onCreate(db: SQLiteDatabase) {
+                Log.i(TAG, "Creating AndBible tables (ReadingPlan)")
+                db.execSQL(SQL_CREATE_ENTRIES_META)
+                db.execSQL(SQL_CREATE_ENTRIES_DAYS)
+
+                importReadingPlansToDatabase(db)
+
+                db.execSQL(SQL_CREATE_DAYS_INDEX)
+            }
+
+            fun importReadingPlansToDatabase(db: SQLiteDatabase, uri: Uri? = null): Boolean {
+                // dbAdapter can ONLY be used in this function if it's not [firstImport]
+                lateinit var dbAdapter: ReadingPlanDBAdapter
+                lateinit var readingPlanProperties: ArrayList<ReadingPlanPropertiesFromText>
+                var planCode: String? = null
+                var customInput: InputStream? = null
+                val firstImport = uri == null
+                if (!firstImport) {
+                    dbAdapter = ReadingPlanDBAdapter()
+                    customInput = BibleApplication.application.contentResolver.openInputStream(uri!!)
+                    planCode = getPlanCodeAsFileName(uri)
+                    if (planCode == null) return false // it's not a .properties file
+                }
+                var returnValue = false
+
+                readingPlanProperties = when (firstImport) {
+                    true -> getAllReadingPlanProperties
```

Comment:
```suggestion
                    true -> allReadingPlanProperties
```

(property naming convention)

Reply:
Done